### ISSUE TITLE
why no packages?

### ISSUE THREAD

**Issue by doctorcolossus** [stevenlovegrove/Pangolin#905]
Hello! I am considering trying to package pangolin in an effort to streamline the installation process of [ORB_SLAM3](https://github.com/UZ-SLAMLab/ORB_SLAM3/tree/master/Examples/Stereo-Inertial). However, I wanted to first ask whether there is any reason that this apparently not already been attempted by anyone. Is there some reason why it is particularly difficult? Would I regret even trying this? Thanks in advance.

**Comment by christian-rauch** [comment]
Do you mean something like a Debian or conda package? I published the `pangolin` package to the ROS repo (https://index.ros.org/p/pangolin/github-stevenlovegrove-Pangolin/). It should be available on `humble` after the next sync. Even without using ROS, you can add the ROS repo and use the binary Pangolin package after sourcing.

If you are interested in a conda package, I started creating a recipe here https://github.com/christian-rauch/staged-recipes/tree/pangolin.

**Comment by doctorcolossus** [comment]
Yep, I was thinking of making a Debian-style `.deb` package for Ubuntu, but my question was just generally whether anyone had made any package anywhere.

I don't really know anything about Pangolin myself except that it's a build dependency of ORB_SLAM3. ORB_SLAM3 is quite a pain to build, and has several other dependencies that also need to be built from source, so I was just thinking about how to simplify that. It seems like mostly people would want to use ORB_SLAM3 with ROS, and ROS is pretty Ubuntu-specific, so it's surprising to me that apparently no one has tried packaging that or its dependencies before.

I was able to build Pangolin eventually, but struggled for a while with:
`ninja: error: unknown target 'pypangolin_pip_install'`
... and:
`# make: *** No rule to make target 'pypangolin_pip_install'.  Stop.`
... before correctly venturing that maybe I didn't need the Python installation to build ORB_SLAM3.
And I just wonder how many other people have had to go through the same thing, and why apparently no one tried to package this or any of the other dependencies up.

But it's good to know about your ROS and conda efforts! Thank you. It looks like the ROS package still needs to be built from source via `catkin`, right? I was wondering about a binary package that could just be `apt install`ed and boom. I also found that someone made [a FreeBSD package](https://pkgs.org/search/?q=pangolin). But it's strange there's none for Ubuntu.

**Comment by christian-rauch** [comment]
With "ROS repo" I meant the bloomed binary Debian packages in the apt repo (http://packages.ros.org/ros2/ubuntu). You can access the package from the ros2-testing pre-release repo: https://docs.ros.org/en/rolling/Installation/Testing.html. In fact, you could also download that one package directly from here: http://packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-humble-pangolin/.

**Comment by christian-rauch** [comment]
I am closing this as Debian packages are now available from the ROS bloom repo. If you need binary packages through other channels, I would suggest asking Linux distributions to package Pangolin.
